src/submission/validation-clinical/donor.ts

```diff
@@ -75,22 +96,171 @@ function checkTimeConflictWithSpecimens(
+    programShortName,
+    entityTypes: [
+      ClinicalEntitySchemaNames.PRIMARY_DIAGNOSIS,
+      ClinicalEntitySchemaNames.TREATMENT,
+      ClinicalEntitySchemaNames.FOLLOW_UP,
+    ],
+    page: 0,
+    sort: 'donorId',
+    donorIds: [],
+    submitterDonorIds: [],
+  };
+
+  const { donors } = await donorDao.findByPaginatedProgramId(programId, query);
+
+  const storedTreatments = donors.map(donor => donor.treatments).filter(donor => Boolean(donor));
+  const storedPrimaryDiagnoses = donors
+    .map(donor => donor.primaryDiagnoses)
+    .filter(donor => Boolean(donor));
+  const storedFollowUps = donors.map(donor => donor.followUps).filter(donor => Boolean(donor));
+
+  const allTreatments = [treatments, submittedTreatments, storedTreatments]
+    .filter(notEmpty)
+    .flat() as Treatment[];
+
+  const allPrimaryDiagnoses = [primaryDiagnoses, submittedPrimaryDiagnoses, storedPrimaryDiagnoses]
+    .filter(notEmpty)
+    .flat() as PrimaryDiagnosis[];
+
+  const allFollowUps = [followUps, submittedFollowUps, storedFollowUps]
+    .filter(notEmpty)
+    .flat() as FollowUp[];
+
+  if (lost_to_followup_after_clinical_event_id) {
+    const primaryDiagnosisMatch = allPrimaryDiagnoses?.find(
+      primaryDiagnosisRecord =>
+        primaryDiagnosisRecord.clinicalInfo?.submitter_primary_diagnosis_id ===
+        lost_to_followup_after_clinical_event_id,
     );
+
+    const treatmentMatch = allTreatments?.find(
+      treatmentRecord =>
+        treatmentRecord.clinicalInfo?.submitter_treatment_id ===
+        lost_to_followup_after_clinical_event_id,
+    );
+
+    const followUpMatch = allFollowUps?.find(
+      followUpRecord =>
+        followUpRecord.clinicalInfo?.submitter_follow_up_id ===
+        lost_to_followup_after_clinical_event_id,
+    );
+
+    const entityIdMatch = primaryDiagnosisMatch || treatmentMatch || followUpMatch;
+
+    if (!entityIdMatch) {
+      errors.push(
+        utils.buildSubmissionError(
+          submittedDonorRecord,
+          DataValidationErrors.INVALID_LOST_TO_FOLLOW_UP_ID,
+          DonorFieldsEnum.lost_to_followup_after_clinical_event_id,
+          {
+            lost_to_followup_after_clinical_event_id,
+          },
+        ),
+      );
+    } else {
+      const { interval_of_followup } = entityIdMatch.clinicalInfo;
+      const submittedInterval =
+        interval_of_followup && typeof interval_of_followup === 'number' ? interval_of_followup : 0;
+
+      const invalidTreatments = allTreatments.filter(treatment => {
+        const { treatment_start_interval, treatment_duration } = treatment.clinicalInfo
+          ? treatment.clinicalInfo
+          : treatment;
+
+        const treatmentInterval =
+          treatment_start_interval && typeof treatment_start_interval === 'number'
+            ? treatment_start_interval
+            : 0;
+
+        const treatmentDuration =
+          treatment_duration && typeof treatment_duration === 'number' ? treatment_duration : 0;
+
```

Comment:
these checks can be shortened because `typeof` will work on pretty much anything.
eg. `typeof this_is_not_a_valid_variable` will return 'undefined'

Reply:
Updated